### DISCUSSION TITLE
Quarto Blog: How to enable sidebar only in posts that have multiple pages?

### DISCUSSION THREAD

**Discussion by InvictusNavarchus** [quarto-dev/quarto-cli#10707] [Q&A] [answered]
### Description

So, I just created a new quarto blog from RStudio. After adding some things, the posts directory looks like this:
```
├── posts
│   ├── _metadata.yml
│   ├── multiple_pages
│   │   ├── index.qmd
│   │   ├── _metadata.yml
│   │   ├── part1.qmd
│   │   └── part2.qmd
│   └── single_page
│       └── index.qmd
```
Inside `multiple_pages` post directory, there are 3 `.qmd` files. I'd like a sidebar to show pages to part 1 and part 2.

After reading https://quarto.org/docs/websites/website-navigation.html#side-navigation, I tried as follow:
I created a new `_metadata.yml` right inside that folder with the following metadata to enable sidebar:

```_metadata.yml
website:
  sidebar:
    style: "docked"
    search: true
    contents:
      - text: "Introduction"
        href: index.qmd
      - text: "Part 1"
        href: part1.qmd
      - text: "Part 2"
        href: part2.qmd
```

Unfortunately, it doesn't work. The sidebar doesn't even appear. 

![Screenshot_20240904_112633](https://github.com/user-attachments/assets/e97c8295-cb81-4d1b-ae01-0791e21af46d)

A second issue is, `part1.qmd` and `part2.qmd` are listed as an independent posts on the homepage:

![Screenshot_20240904_112739](https://github.com/user-attachments/assets/eaaca958-2068-444b-82ef-15a1c15b783a)

Here are the links to both the repo and the live page:
- Repository: https://github.com/InvictusNavarchus/quarto_create_sidebar_inside_posts
- Live Page: https://navarchus.id/quarto_create_sidebar_inside_posts/

Note: This issue is similar to #9782

**Comment by mcanouil** [comment, marked as answer]
All sidebars must be defined in `_quarto.yml` and referenced (so you need to set `id` as shown in the documentation) where you want it.

**Reply by InvictusNavarchus** [reply]
@mcanouil Ahh, I see, but there's only 1 `_quarto.yml`. Does it mean I have to put all the sidebars of all the posts inside that one `_quarto.yml`?

**Reply by mcanouil** [reply]
Yes, that is what I said.

**Reply by InvictusNavarchus** [reply]
Got it. I just tested it and it worked as I wanted. Although, I might need to do something to refactor the `_quarto.yml` somehow because there's going to be dozens of posts.

Anyway, thanks for the quick response  :+1:

**Reply by InvictusNavarchus** [reply]
Also, it's not clear in the documentation on how to set the `id`.  Thankfully, someone explained it in #8117. I'll raise an issue on it if not already.

**Reply by mcanouil** [reply]
What do you mean "set the id"?
Add `id: value` as all listed keys for sidebar in the documentation.

If you meant how to reference the id, yes. I'll make a PR to add a note about this.

**Reply by InvictusNavarchus** [reply]
Well, in #8117, @dragonstyle mentioned that you need to define the `sidebar` metadata in the page you want the sidebar to appear. The value of `sidebar` you set there will be the value you'll reference to in the sidebar's `id` on `.quarto.yml`. This isn't explained in the documentation.

**Reply by InvictusNavarchus** [reply]
(replying to your edit) Yes, exactly, how to reference the id.

**Reply by mcanouil** [reply]
Ok, thanks for the feedback.
I'll make the necessary update to the documentation.

**Reply by InvictusNavarchus** [reply]
No probs. Thank you too.

**Comment by mcanouil** [comment]
Pull request is up to add an example on how to reuse a sidebar:

- https://github.com/quarto-dev/quarto-web/pull/1329

**Reply by InvictusNavarchus** [reply]
Thanks!
